src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -255,136 +252,102 @@
   [insets translate-y]
   [:f> f-bottom-view insets translate-y])
 
-(defn- check-qr-code-data
-  [event]
+(defn- check-qr-code-and-navigate
+  [event set-qr-code-succeeded]
   (let [connection-string        (string/trim (oops/oget event "nativeEvent.codeStringValue"))
         valid-connection-string? (sync-utils/valid-connection-string? connection-string)]
     (if valid-connection-string?
-      (debounce/debounce-and-dispatch [:syncing/input-connection-string-for-bootstrapping
-                                       connection-string]
-                                      300)
+      (do
+        (set-qr-code-succeeded)
```

Comment:
`set-qr-code-succeeded` is a function that will change a control atom, we need it to not scanning the qr code more than once.

The behaviour is similar to the  previous `read-qr-once?`  atom, but now it is handled by the `compnentDidAppear` method